[BUG] v7.0.0 typescript-axios generator generates export interface ProtobufAny different from v6.6.0

#### Bug Report Checklist

- [x] Have you provided a full/minimal spec to reproduce the issue?
- [x] Have you validated the input using an OpenAPI validator ([example](https://apidevtools.org/swagger-parser/online/))?
- [ ] Have you [tested with the latest master](https://github.com/OpenAPITools/openapi-generator/wiki/FAQ#how-to-test-with-the-latest-master-of-openapi-generator) to confirm the issue still exists?
- [x] Have you searched for related issues/PRs?
- [x] What's the actual output vs expected output?
- [ ] [Optional] Sponsorship to speed up the bug fix or feature request ([example](https://github.com/OpenAPITools/openapi-generator/issues/6178))

<!--
Please follow the issue template below for bug reports.
Also please indicate in the issue title which language/library is concerned. Eg:  [BUG][JAVA] Bug generating foo with bar 
-->

##### Description

<!-- describe what is the question, suggestion or issue and why this is a problem for you. -->

v6.6.0 generates what we and tsc expected, v7.0.0 doesn't.


##### openapi-generator version

<!-- which version of openapi-generator are you using, is it a regression? -->

v7.0.0

##### OpenAPI declaration file content or url

<!-- if it is a bug, a json or yaml that produces it.
If you post the code inline, please wrap it with
```yaml
(here your code)
```
(for YAML code) or
```json
(here your code)
```
(for JSON code), so it becomes more readable. If it is longer than about ten lines,
please create a Gist (https://gist.github.com) or upload it somewhere else and
link it here.
  -->


swagger spec file t.json

```json
{
  "swagger": "2.0",
  "info": {
    "title": "foo APIs",
    "version": "0.0.1"
  },
  "host": "foo.com",
  "paths": {
    "/hello": {}
  },
  "definitions": {
    "protobufAny": {
      "type": "object",
      "properties": {
        "@type": {
          "type": "string"
        }
      },
      "additionalProperties": {}
    }
  }
}

```


openapi tool configuration file foo.openapitools.json
```json
{
  "npmName": "foo",
  "npmVersion": "0.0.1",
  "snapshot": false,
  "supportsES6": true,
  "modelPropertyNaming": "original"
}
```



##### Generation Details

<!-- 
    Prefer CLI steps, including the language, libraries and various options. 
    Providing config-based settings allows for simpler testing across CLI and plugins. 
    For examples, see https://github.com/OpenAPITools/openapi-generator/tree/master/bin/configs
-->

##### Steps to reproduce

<!-- unambiguous set of steps to reproduce the bug.-->



generate code from spec

```shell
openapi-generator-cli version-manager set 7.0.0
openapi-generator-cli generate -i t.json --generator-name typescript-axios -o src/gen/foo -c foo.openapitools.json
```

tsc build failed with v7.0.0

```shell
npm run --prefix src/gen/foo build                                                                                

> foo@0.0.1 build
> tsc && tsc -p tsconfig.esm.json

api.ts:39:5 - error TS2411: Property ''@type'' of type 'string' is not assignable to 'string' index type 'object'.

39     '@type'?: string;
       ~~~~~~~


Found 1 error in api.ts:39
```




##### Related issues/PRs

<!-- has a similar issue/PR been reported/opened before? Please do a search in https://github.com/openapitools/openapi-generator/issues?utf8=%E2%9C%93&q=is%3Aissue%20 -->

##### Suggest a fix

<!-- if you can't fix the bug yourself, perhaps you can point to what might be
  causing the problem (line of code or commit), or simply make a suggestion -->

If we switch back to 6.6.0 everything works as expected:

```shell
openapi-generator-cli version-manager set 6.6.0

openapi-generator-cli generate -i t.json --generator-name typescript-axios -o src/gen/foo -c foo.openapitools.json
npm run --prefix src/gen/foo build
```

we diff v6.6.0 and v7.0.0 

```diff
 diff -Nuar  src/gen/foo/api.ts /tmp/app/src/gen/foo/api.ts
--- src/gen/foo/api.ts	2023-09-04 13:39:42
+++ /tmp/app/src/gen/foo/api.ts	2023-09-04 13:39:57
@@ -29,7 +29,7 @@
  * @interface ProtobufAny
  */
 export interface ProtobufAny {
-    [key: string]: object | any;
+    [key: string]: object;
 
     /**
      * 
```

--- a/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java
+++ b/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/TypeScriptAxiosClientCodegen.java
@@ -226,6 +226,11 @@ public class TypeScriptAxiosClientCodegen extends AbstractTypeScriptClientCodege
         for (ModelMap mo  : models) {
             CodegenModel cm = mo.getModel();
 
+            // Type is already any
+            if (cm.getAdditionalPropertiesIsAnyType() && "any".equals(cm.getAdditionalPropertiesType())) {
+                cm.setAdditionalPropertiesIsAnyType(false);
+            }
+
             // Deduce the model file name in kebab case
             cm.classFilename = cm.classname.replaceAll("([a-z0-9])([A-Z])", "$1-$2").toLowerCase(Locale.ROOT);
 

--- a/modules/openapi-generator/src/main/resources/typescript-axios/modelGeneric.mustache
+++ b/modules/openapi-generator/src/main/resources/typescript-axios/modelGeneric.mustache
@@ -5,7 +5,7 @@
  */
 export interface {{classname}} {{#parent}}extends {{{.}}} {{/parent}}{
 {{#additionalPropertiesType}}
-    [key: string]: {{{additionalPropertiesType}}}{{^additionalPropertiesIsAnyType}}{{#hasVars}} | any{{/hasVars}}{{/additionalPropertiesIsAnyType}};
+    [key: string]: {{{additionalPropertiesType}}}{{#additionalPropertiesIsAnyType}}{{#hasVars}} | any{{/hasVars}}{{/additionalPropertiesIsAnyType}};
 
 {{/additionalPropertiesType}}
 {{#vars}}

--- a/samples/client/petstore/typescript-axios/builds/test-petstore/api.ts
+++ b/samples/client/petstore/typescript-axios/builds/test-petstore/api.ts
@@ -447,7 +447,7 @@ export interface Dog extends Animal {
  * @interface Drawing
  */
 export interface Drawing {
-    [key: string]: Fruit;
+    [key: string]: Fruit | any;
 
     /**
      * 
@@ -1043,7 +1043,7 @@ export interface Name {
  * @interface NullableClass
  */
 export interface NullableClass {
-    [key: string]: object;
+    [key: string]: object | any;
 
     /**
      * 

--- a/samples/client/petstore/typescript-axios/builds/with-fake-endpoints-models-for-testing-with-http-signature/api.ts
+++ b/samples/client/petstore/typescript-axios/builds/with-fake-endpoints-models-for-testing-with-http-signature/api.ts
@@ -811,7 +811,7 @@ export interface Name {
  * @interface NullableClass
  */
 export interface NullableClass {
-    [key: string]: object;
+    [key: string]: object | any;
 
     /**
      * 

